### PR TITLE
First go at adding esutil build package

### PR DESCRIPTION
**Pull request by timj**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +0,0 @@
-# esutil
```
Comment (timj) [inline README.md:1]: Can you put the README back (preferably with more text)? I imagine that needs `TAP_PACKAGE=1` to go in an `eupspkg.cfg.sh` file.

File: ups/esutil.table [hunk 2]
```diff
@@ -0,0 +1,5 @@
+setupRequired(python)
+setupRequired(numpy)
+envPrepend(PYTHONPATH, ${PRODUCT_DIR}/lib/python)
+envPrepend(LD_LIBRARY_PATH, ${PRODUCT_DIR}/lib)
+envPrepend(DYLD_LIBRARY_PATH, ${PRODUCT_DIR}/lib)
```
Comment (timj) [inline ups/esutil.table]: I don't think these `LD_LIBRARY_PATH` lines should be here.

**Comment by timj** [comment]
How are we going to handle the problem that `esutil` has its own `ups` directory and completely overrides the table file that is included here and has different stuff in it? At the very least a comment needs to be made that the table file in the repository will be replaced.

**Comment by timj** [comment]
Does `esutil` have any tests that I can run to see if it works?

**Comment by timj** [comment]
`esutil` depends on scipy so we probably need to fix that when lsst/scipy#1 is merged (assuming we can work out how to keep our stuff in the `ups` file).

**Comment by SimonKrughoff** [comment]
I was pretty happy when it all just worked, but all these are good points.  Do you have a suggestion regarding the `ups` directory in `esutil`?

**Comment by timj** [comment]
Since patches are applied at prep time and the table file is over written at build time I think the best option is to have a patch file that rewrites `setup.py` to do what we want it to do.

File: ups/esutil.table (hunk 2 above)
Comment (SimonKrughoff) [inline ups/esutil.table]: good call.  Cargo culted.

**Comment by SimonKrughoff** [comment]
There are no tests that I know of.  Is that a blocker (even if this is meant to be temporary)?

**Comment by timj** [comment]
It's not a blocker if upstream don't have tests. I just wanted to find out if my local build actually worked...

**Comment by timj** [comment]
One complication is that `esutil` has its own table file that is copied in during the `prep` phase from the tar file. This replaces the version you have written. A patch file that solely prevents `setup.py` from overwriting the table file is too late.

**Comment by SimonKrughoff** [comment]
@timj very sorry but I think I force pushed and erased your updates to the README.md.

If you are going to make more edits, give me a second, because I need to rebase again.

**Comment by SimonKrughoff** [comment]
O.K. It actually works now.  It didn't before, but I didn't notice because I had esutil installed in my local environment.  I've checked and it imports and from the correct location.

I got around the table file problem by modifying the tarball in `upstream` to remove the `esutil.table` that ships with the package.  I also added a patch that takes out the code to overwrite the table file.  I hope that's sufficient.

**Comment by timj** [comment]
@SimonKrughoff I don't remember pushing any changes to the readme...

**Comment by SimonKrughoff** [comment]
O.K.  I didn't confirm that it was you.  I just saw some changes go by, but didn't realize the ramification until after I pushed.

**Pull request merged**